core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -932,7 +932,10 @@ public ClusterState execute(ClusterState currentState) {
                     if (modified) {
                         stateBuilder.nodes(nodesBuilder);
                     }
-                    return stateBuilder.build();
+                    currentState = stateBuilder.build();
```

Comment:
I was looking to reduce the amount of logical change here, but I can make this change...